code/_helpers/mobs.dm

```diff
@@ -284,4 +284,12 @@ proc/age2agedescription(age)
 		res += "dispersed"
 	if(damflags & DAM_BIO)
 		res += "bio"
-	return english_list(res)
\ No newline at end of file
+	return english_list(res)
+
+/proc/get_effective_view(var/val)
+	if(isnum(val))
+		return val
+	if(istext(val))
+		var/list/vals = splittext(val, "x")
+		return min(text2num(vals[1]), text2num(vals[2]))
```

Comment:
If views are centered on players, then technically anything "in view" is the half of the largest dimension.

Reply:
Tweaked accordingly, cheers